### PR TITLE
Try to add Strategy Pattern

### PR DESCRIPTION
**Pull request by Blackolf931**

### CODE REVIEW

File: TSP/BLL/Models/EmployeeGetById.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace BLL.Models
+{
+    public class EmployeeGetById
```
Comment (nantonov) [inline TSP/BLL/Models/EmployeeGetById.cs:8]: Strange name

**Review by nantonov — CHANGES_REQUESTED** [review]

File: TSP/BLL/Services/EmployeeService.cs [hunk 2]
```diff
@@ -11,11 +11,13 @@ public class EmployeeService : IEmployeeService
     {
         private readonly IEmployeeRepository _repository;
         private readonly IMapper _mapper;
+        private readonly IEnumerable<IStrategyService> listStrategy;
```
[the remark this answers is not in the document]
Comment (Blackolf931) [reply TSP/BLL/Services/EmployeeService.cs]: fix

File: TSP/BLL/Services/EmployeeService.cs [hunk 3]
```diff
@@ -36,10 +37,19 @@ public async Task<IEnumerable<Employee>> GetAllAsync()
             return _mapper.Map<IEnumerable<Employee>>(employee);
         }
 
-        public async Task<Employee> GetEmployeeByIdAsync(int id)
+        public async Task<EmployeeGetById> GetEmployeeByIdAsync(int id)
```
[the remark this answers is not in the document]
Comment (Blackolf931) [reply TSP/BLL/Services/EmployeeService.cs]: fix

File: TSP/BLL/Services/GetAdditionalInformationAboutRetirePeople.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+namespace BLL.Services
+{
+    class GetAdditionalInformationAboutRetirePeople : IStrategyService
+    {
+        public string SetInformation()
+        {
+            return new StringBuilder().Append("Your are retire").ToString();
+        }
+
+        bool IStrategyService.IsValidStrategy(int age)
```
[the remark this answers is not in the document]
Comment (Blackolf931) [reply TSP/BLL/Services/GetAdditionalInformationAboutRetirePeople.cs]: fix

File: TSP/BLL/Services/IGetAdditionalInterface.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BLL.Services
+{
+    public interface IGetAdditionalInterface
```
[the remark this answers is not in the document]
Comment (Blackolf931) [reply TSP/BLL/Services/IGetAdditionalInterface.cs]: deleted

File: TSP/BLL/Services/IStrategyService.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BLL.Services
+{
+    public interface IStrategyService
```
[the remark this answers is not in the document]
Comment (Blackolf931) [reply TSP/BLL/Services/IStrategyService.cs]: fix

File: TSP/BLL/Services/StrategyService.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BLL.Services
+{
+    class StrategyService
```
[the remark this answers is not in the document]
Comment (Blackolf931) [reply TSP/BLL/Services/StrategyService.cs]: deleted

File: TSP/DAL/Repositories/EmployeeRepository.cs [hunk 8]
```diff
@@ -52,7 +52,7 @@ public async Task<EmployeeEntity> GetByIdAsync(int id)
         public async Task<EmployeeEntity> UpdateAsync(EmployeeEntity entity)
         {
            _repositoryContext.Employees.Update(entity);
-           _repositoryContext.Entry(entity).State = EntityState.Unchanged;
+        //   _repositoryContext.Entry(entity).State = EntityState.Unchanged;
```
[the remark this answers is not in the document]
Comment (Blackolf931) [reply TSP/DAL/Repositories/EmployeeRepository.cs]: fix

File: TSP/BLL/Models/EmployeeGetById.cs (hunk 1 above)
Comment (Blackolf931) [reply TSP/BLL/Models/EmployeeGetById.cs]: deleted

File: TSP/BLL/DI/DependencyRegistrar.cs [hunk 9]
```diff
@@ -16,7 +16,10 @@ public static void RegistarBuisnessComponents(this IServiceCollection services,
             services.AddScoped<IOfficeService, OfficeService>();
             services.AddScoped<IEmployeeRepository, EmployeeRepository>();
             services.AddScoped<IOfficeRepository, OfficeRepository>();
-            
+            services.AddScoped<IStrategy, GetAdditionalInformationAboutRetirePeople>();
+            services.AddScoped<IStrategy, GetAdditionalInformationAboutMiddlePeople>();
+            services.AddScoped<IStrategy, GetAdditionalInformationAboutYoungPeople>();
```
Comment (nantonov) [inline TSP/BLL/DI/DependencyRegistrar.cs:21]: rename pls

**Review by nantonov — APPROVED** [review]

File: TSP/BLL/DI/DependencyRegistrar.cs (hunk 9 above)
Comment (Blackolf931) [reply TSP/BLL/DI/DependencyRegistrar.cs]: fix

**Review by nantonov — APPROVED** [review]

**Pull request merged**